src/vm/comcallablewrapper.h

```diff
@@ -402,46 +426,52 @@ class ComCallWrapperTemplate
 
     IIDToInterfaceTemplateCache *GetOrCreateIIDToInterfaceTemplateCache();
 
+    BOOL HasInvisibleParent()
+    {
+        LIMITED_METHOD_CONTRACT;
+        return HasFlag(m_flags, ComCallWrapperTemplateFlags::InvisibleParent);
+    }
+
     BOOL SupportsICustomQueryInterface()
     {
         LIMITED_METHOD_CONTRACT;
-        return (m_flags & enum_ImplementsICustomQueryInterface);
+        return HasFlag(m_flags, ComCallWrapperTemplateFlags::ImplementsICustomQueryInterface);
     }
 
     BOOL SupportsIInspectable()
     {
         LIMITED_METHOD_CONTRACT;
-        return (m_flags & enum_SupportsIInspectable);
+        return HasFlag(m_flags, ComCallWrapperTemplateFlags::SupportsIInspectable);
     }
 
     BOOL SupportsVariantInterface()
     {
         LIMITED_METHOD_CONTRACT;
-        return (m_flags & enum_SupportsVariantInterface);
+        return HasFlag(m_flags, ComCallWrapperTemplateFlags::SupportsVariantInterface);
     }
 
     BOOL RepresentsVariantInterface()
     {
         LIMITED_METHOD_CONTRACT;
-        return (m_flags & enum_RepresentsVariantInterface);
+        return HasFlag(m_flags, ComCallWrapperTemplateFlags::RepresentsVariantInterface);
     }
 
     BOOL IsUseOleAutDispatchImpl()
     {
         LIMITED_METHOD_CONTRACT;
-        return (m_flags & enum_UseOleAutDispatchImpl);
+        return HasFlag(m_flags, ComCallWrapperTemplateFlags::UseOleAutDispatchImpl);
```

Comment:
Why is `HasFlag` method better than a simple `&` ?

Reply:
I converted the `enum` to an `enum class` which doesn't degrade to an `int`. This means that `x & y` cannot degrade further to a predicate expression. The 'improvement' here was further type safety. By using `enum` there is no type enforcement since any integral type will degrade to be that a member of that `enum` - an unfortunate side effect of type safety is that a `HasFlag()` type of function needs to be created.

Reply:
To specifically answer your question - we get type safety during compilation and the optimizer will elide the function entirely during a release build to it is safer with equal performance.

Reply:
You are adding a bit of safety on one side, and removing safety on the other side by adding `(DWORD*)` casts that can result into much harder to trace down bugs.

Reply:
To be fair it is also about readability. The `DWORD m_flags` doesn't tell me anything about the field, properly typing it indicates purpose and use. I agree the interlocked OR cast isn't great and I would like to address that, but I don't want to touch that function since it is used in many places. I can declare the enumeration as `enum class ComCallWrapperTemplateFlags : DWORD`, even then a cast would be needed, but then the cast could be type checked.

Reply:
I think readability is about consistency accross the codebase. This is yet another style to deal with bitfields enums. Bit field enums work poorly in C++. IMHO, there is no actually good solution to deal with them.

The style frequently used to address the readability problem in this codebase is to put the enum definition right in front of the field, like here: https://github.com/dotnet/coreclr/blob/master/src/vm/method.hpp#L1715.

Reply:
Yes, I would keep original styple. since it is widely used in coreclr code.